carto_report/report.py

```diff
@@ -24,614 +25,595 @@
+        names_positions = [i for i, _ in enumerate(analysis_names)]
+
+        # create plot
+        fig_analysis, ax_analysis = plt.subplots()
+
+        # plot bars
+        ax_analysis.barh(names_positions, analysis_portions, color=cartocolors)
+
+        # customize ticks and labels
+        ax_analysis.set_ylabel("Analysis Type")
+        ax_analysis.set_xlabel("Analysis Count")
+        ax_analysis.set_yticks(names_positions)
+        ax_analysis.set_yticklabels(analysis_names)
+
+        # tight plot
+        plt.tight_layout()
+
+        return fig_analysis
+
+    ### generate report with an HTML template
+
+    def generateReport(self,
+        user, org, today, 
+        lds_df, 
+        maps_df, top_5_maps_date,
+        analysis_types_df, analysis_df,
+        dsets_df, tables_sizes, top_5_dsets_date, top_5_dsets_size,
+        sync, private, link, public,
+        geo, none_tbls, points, lines, polys,
+        fig_analysis,fig_lds):
+
+        '''
+        Method to generate a HTML report.
+        '''
+
+        self.logger.info('Generating HTML template...')
+
+        template = """
+            <!DOCTYPE html>
+            <html lang="en">
+            <head>
+            <meta charset="UTF-8">
+            <meta name="viewport" content="width=device-width, initial-scale=1.0">
+            <meta http-equiv="X-UA-Compatible" content="ie=edge">
+            <title>CARTO Database Metrics Report Template</title>
+            <link rel="stylesheet" href="https://libs.cartocdn.com/airship-style/v1.0.3/airship.css">
+            <script src="https://libs.cartocdn.com/airship-components/v1.0.3/airship.js"></script>
+            <style>
+            .as-sidebar{
+                width: 33.33%;
+            }
+            .as-box{
+                border-bottom: 1px solid #F5F5F5;
+            }
+            </style>
+            </head>                   
+            <body class="as-app-body as-app">
+            <header class="as-toolbar">
+                <div class="as-toolbar__item as-title">
+                    CARTO Metrics Report 
+                </div>
+                <div class="as-toolbar__item as-display--block as-p--12 as-subheader as-bg--complementary">
+                    {{ user }} from {{org}} at {{today}}
+                </div>
+            </header>
+            <div class="as-content">
+                <aside class="as-sidebar as-sidebar--left">
+                <div class="as-container">
+                    <h1 class="as-box as-title as-font--medium">
+                    Maps and Analysis
+                    </h1>
+                    <div class="as-box">
+                        <h2 class="as-title">
+                            Maps
+                        </h2>
+                        <p class="as-body as-font--medium">Number of maps: {{total_maps}}</p>
+                        <div class="as-box" id="maps-table">
+                            {{top_5_maps_date.to_html()}}
```

Comment:
It's a pity that the maps urls here are not clickable on this table. I understand you use the convenient `to_html()` function and we can leave it as an enhancement for next version.

Reply:
Yes, that would be great. But we can leave this feature for the next iteration.